hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/BasicValuedCollectionPart.java

```diff
@@ -57,21 +59,32 @@
 
 	private final String tableExpression;
 	private final String columnExpression;
+	private final String customColumnReadExpr;
+	private final String customColumnWriteExpr;
 
 	public BasicValuedCollectionPart(
 			CollectionPersister collectionDescriptor,
 			Nature nature,
 			JdbcMapping mapper,
 			BasicValueConverter valueConverter,
 			String tableExpression,
-			String columnExpression) {
+			String columnExpression,
+			String customColumnReadExpr,
+			String customColumnWriteExpr) {
 		this.navigableRole = collectionDescriptor.getNavigableRole().append( nature.getName() );
 		this.collectionDescriptor = collectionDescriptor;
 		this.nature = nature;
 		this.mapper = mapper;
 		this.valueConverter = valueConverter;
 		this.tableExpression = tableExpression;
 		this.columnExpression = columnExpression;
+
+		this.customColumnReadExpr = customColumnReadExpr == null
+				? columnExpression
+				: customColumnReadExpr;
+		this.customColumnWriteExpr = customColumnWriteExpr == null
+				? DEFAULT_COLUMN_WRITE_EXPRESSION
+				: customColumnWriteExpr;
```

Comment:
We might well use the util method of `org.apache.commons.lang3.ObjectUtils#defaultIfNull()` to improve readability. The class seems present in the classpath.

Reply:
It is not part of the dependencies for hibernate-core.  The only usage of this package in the entire project is in the documentation project.  So we should not use that.

Actually though, looking at this again, this should not be doing this anyway.  It should be using the passed values strictly.